src/master-playlist-controller.js

```diff
@@ -944,9 +953,15 @@ export class MasterPlaylistController extends videojs.EventTarget {
     let newDuration = Hls.Playlist.duration(this.masterPlaylistLoader_.media());
     let buffered = this.tech_.buffered();
     let setDuration = () => {
-      this.mediaSource.duration = newDuration;
+      // on firefox setting the duration may sometimes cause an exception
+      // even if the media source is open and source buffers are not
+      // updating.
+      try {
+        this.mediaSource.duration = newDuration;
+      } catch (e) {
+        videojs.log.warn('Failed to set media source duration', e);
```

Comment:
I tried for a long time to determine the cause of this, it almost seems like a bug in firefox. Originally I thought it was due to `updating` sourceBuffers or the media source not being `open` but that is not the case.